interfaces/devdoc/sync_requirements.md

```diff
@@ -0,0 +1,42 @@
+# sync
+================
+
+## Overview
+
+`sync` provides platform-independent synchronization primitives.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout);
+MOCKABLE_FUNCTION(, void, wake_by_address_all, void*, address);
+MOCKABLE_FUNCTION(, void, wake_by_address_single, void*, address);
+```
+
+## wait_on_address
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout)
+```
+

```

Comment:
Typically, each function should have a brief description before the specs (e.g. see interlocked_hl_requirements.md). This could briefly state what the function does in one sentence, along with any useful notes.